xbmc/filesystem/PluginDirectory.cpp

```diff
@@ -486,35 +485,26 @@ bool CPluginDirectory::WaitOnScriptResult(const std::string &scriptPath, int scr
       CScriptObserver scriptObs(scriptId, m_fetchComplete);
       if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
       {
-        cancelled = true;
+        m_cancelled = true;
       }
       scriptObs.Abort();
     }
   }
   else
   {
-    // kill the script if it does not return within 30 seconds
-    // and we are not waiting for user input
+    // Wait for directory fetch to complete, end, or be cancelled
+    while (!m_cancelled
```

Comment:
I have been thinking that malfunctioning scripts can be killed by the app. I did not observe any issues with killed scripts during my tests. What are you concerns?